File: src/main/java/step1/ConsoleStatement.java
package step1;

public class ConsoleStatement {

	public String statement(Customer customer) {
		return  statementHeader(customer) 
				+ showFigures(customer) 
				+ statementTail(customer);
	}

	private String statementTail(Customer customer) {
		String result = "Amount owed is " + customer.getTotalAmount() + "\n";
		result += "You earned " + customer.getFrequentRenterPoints() + " frequent renter pointers";
		return result;
	}

	private String showFigures(Customer customer) {
		String result = "";
		for ( Rental rental : customer.getRentals() ) {
			result += "\t" +  String.valueOf(rental.getPrice()) + "(" + rental.getMovie().getTitle() + ")" + "\n";
		}
		return result;
	}

	private String statementHeader(Customer customer) {
		return "Rental Record for " + customer.getName() + "\n";
	}

}

File: src/main/java/step1/ConsoleStatement.kt
package step1

class ConsoleStatement {
    fun getStatement(customer: Customer): String {
        return header(customer) + body(customer) + tail(customer)
    }

    private fun header(customer: Customer) = customer.header

    private fun body(customer: Customer) = customer.figures

    private fun tail(customer: Customer) = customer.tail
}
